ChangeLog.d/announce-4.0-removals.txt

```diff
@@ -0,0 +1,29 @@
+New deprecations
+   * The following cryptographic mechanisms are planned to be removed
+     in Mbed TLS 4.0:
+     - DES (including 3DES).
+     - PKCS#1v1.5 encryption (RSA-PKCS1-v1_5). (OAEP, PSS, and PKCS#1v1.5
+       signature are staying.)
+     - Finite-field Diffie-Hellman with custom groups. (RFC 7919 remain
+       supported.)
+     - Elliptic curves of size 225 bits or less.
+   * The following mechanisms are planned to be removed from (D)TLS 1.2
+     in Mbed TLS 4.0:
+     - RSA decryption (i.e. cipher suites using RSA without a key exchange:
+       cipher suites using an RSA signature and ECDHE are staying).
+     - Static ECDH (ephemeral ECDH, i.e. cipher suites using ECDHE, is staying).
+     - Finite-field Diffie-Hellman (i.e. DHE; ECDHE is staying)
+     - All cipher suites using CBC.
+   * The following low-level interfaces are planned to be removed from the
+     public API in Mbed TLS 4.0:
+     - Hashes: md5.h, ripemd160.h, sha1.h, sha3.h, sha256.h, sha512.h;
+     - Pseudorandom generation: ctr_drbg.h, hmac_drbg.h.
+     - Cipher primitives: aes.h, aria.h, camellia.h, chacha20.h,
+       chachapoly.h, poly1305.h;
```

Comment:
Minor: I find the distinction between this line and the line below a bit unclear, perhaps make it a single item "Ciphers and related"? (Also, I wouldn't normally describe HKDF as a cipher mode.) Also, should we point out that `nist_kw.h` is staying?

Reply:
Agree with hkdf. I don't feel strongly about separating cipher primitives from modes, so ok. I don't see why we should mention interfaces that are staying here.

Reply:
> I don't see why we should mention interfaces that are staying here.

In general I agree, but since NIST KW is the only "cipher-related thing" that stays, I thought it might be worth pointing it out - as opposed to people who care needing to double-check it wasn't in the list.

But if you disagree I won't insist.